### PR TITLE
Za create favorites button

### PR DESCRIPTION
**Pull request by ZachariasAtwood**
The favorites logic now uses 2 api's: 1 for adding a favorite (in song page) and 1 for updating the favorites hymnal (in hymns list).

### CODE REVIEW

File: server.ts [hunk 1]
```diff
@@ -14,81 +14,106 @@ const sqlite = require('sqlite');
...
+              console.log(err)
+            }
+
+          })
+
+        if (isInDatabase === 0) {
+          console.log("No such user in database")
+          await db.all(`INSERT INTO users(id, email) VALUES('${numberOfUsers + 1}', '${email}');`)
+          console.log("New user added")
+          const favorites = await db.all(`SELECT favorites FROM users WHERE (email = '${email}');`)
```
Comment (michael-hewitt) [inline server.ts:55]: I am curious about the difference between `db.each` and `db.all`?

File: server.ts [hunk 2]
```diff
@@ -14,81 +14,106 @@ const sqlite = require('sqlite');
...
+          console.log("Login Send Favorites: " + data)
+          response.send(data)
+
+        } else if (isInDatabase === 1) {
+          console.log("User found")
+          const favorites = await db.all(`SELECT favorites FROM users WHERE (email = '${email}');`)
+          console.log("Login Favorites: " + favorites)
+          const data = JSON.stringify(favorites, null, 4)
+          console.log("Login Send Favorites: " + data)
+          response.send(data)
```
Comment (michael-hewitt) [inline server.ts:67]: Do we still need to send back the favorites during `/login` now that we also. have `/getFavoritesHymnal`?

File: server.ts [hunk 3]
```diff
@@ -14,81 +14,106 @@ const sqlite = require('sqlite');
...
-        [email],
-        (err, row) => {
+      response.send(true)
+    } else {
+      console.log("JWT Not Working")
+      response.send(false)
+    }
+  })
+
+  app.post('/getFavoritesHymnal', async (request, response) => {
```
Comment (michael-hewitt) [inline server.ts:97]: Perhaps just `/getFavorites`, and it is the client that converts the favorites into a virtual hymnal?

File: src/pages/hymns_list.tsx [hunk 4]
```diff
@@ -59,13 +69,62 @@ export const HymnsList = (props: Props) => {
...
+      if (credentials) {
+        fetch('/getFavoritesHymnal', {
+          method: 'POST',
+          headers: {
+            'Authorization' : `${credentials}`,
+            'Content-Type': 'application/json'
+          },
+        }).then(response => response.json())
+          .then(data => {
+            console.log(Object.values(data[0]))
```
Comment (michael-hewitt) [inline src/pages/hymns_list.tsx:92]: We'll eventually want to remove all these logging statements, but it is fine to leave them in for now.

File: src/pages/hymns_list.tsx [hunk 5]
```diff
@@ -59,13 +69,62 @@ export const HymnsList = (props: Props) => {
...
+            }
+          })
+      } else {
+        setAllHymnals(hymnalsWithoutFavorites.concat([{name: 'Favorites', message: 'Login to choose favorites.'}]))
+      }
+    }
+    console.log("UseEffect has been run.")
+  }, [favorites, credentials, hymnalsWithoutFavorites]);
+
+  console.log(allHymnals)
```
Comment (michael-hewitt) [inline src/pages/hymns_list.tsx:127]: Nice work adding in the artificial Favorites hymnal!

File: src/pages/hymns_list.tsx [hunk 6]
```diff
@@ -236,6 +305,13 @@ export const HymnsList = (props: Props) => {
     setDisplayHymn(undefined);
   }, []);
 
+
+  function setNewFavorites (hymn: Hymn) {
+    console.log("Callback works")
+    setFavorites(hymn)
+  }
```
Comment (michael-hewitt) [inline src/pages/hymns_list.tsx:312]: I'm wondering about how the downloading of favorites & the adding of a new favorite should interact with each other.  Seems like they are messing with the same list somehow.

**Review by michael-hewitt — APPROVED** [review]
Solid progress!

File: server.ts (hunk 1 above)
Comment (ZachariasAtwood) [reply server.ts:55]: I believe they work the same way, only db.each takes a callback function as 2nd parameter.

File: server.ts (hunk 2 above)
Comment (ZachariasAtwood) [reply server.ts:67]: No, you're right, we don't.

**Pull request merged**